compiler/qsc_parse/src/stmt.rs

```diff
@@ -111,15 +111,20 @@ fn parse_qubit(s: &mut Scanner) -> Result<Box<StmtKind>> {
     let lhs = pat(s)?;
     token(s, TokenKind::Eq)?;
     let rhs = parse_qubit_init(s)?;
-    let block = opt(s, parse_block)?;
+    let block = if !s.contains_language_feature(LanguageFeature::V2PreviewSyntax) {
```

Comment:
nitpick: swapping the then-else blocks will let you drop the `!` and improves readability, in my opinion.